src/api/ledger/parse/utils.js

```diff
@@ -4,6 +4,15 @@ const _ = require('lodash');
 const transactionParser = require('ripple-lib-transactionparser');
 const toTimestamp = require('../../../core/utils').toTimestamp;
 const utils = require('../utils');
+const BigNumber = require('bignumber.js');
+
+function adjustQualityForXRP(quality: string, takerGetsCurrency: string,
+    takerPaysCurrency: string) {
+  const shift = (takerGetsCurrency === 'XRP' ? 1 : 0)
```

Comment:
Now fixed, including changes to fixtures that were using it. I didn't use the nested ternary because that seems just as convoluted and doesn't give the expected result for XRP/XRP, which doesn't occur, but can still be misleading to the reader.